src/components/recurring-donation/grid/Grid.tsx

```diff
@@ -1,86 +1,129 @@
 import React, { useState } from 'react'
 import { UseQueryResult } from '@tanstack/react-query'
 import { useTranslation } from 'next-i18next'
-import { Box } from '@mui/material'
+import { IconButton, Tooltip, Box } from '@mui/material'
 import { DataGrid, GridColDef, GridColumns, GridRenderCellParams } from '@mui/x-data-grid'
 
-import { ModalStore } from 'stores/dashboard/ModalStore'
-import { routes } from 'common/routes'
 import { RecurringDonationResponse } from 'gql/recurring-donation'
-import { useRecurringDonationList } from 'common/hooks/recurringDonation'
-import GridActions from 'components/admin/GridActions'
+import { useAllRecurringDonations } from 'common/hooks/recurringDonation'
+import { money } from 'common/util/money'
+import { useMutation } from '@tanstack/react-query'
+import { useRouter } from 'next/router'
+import { endpoints } from 'service/apiEndpoints'
+import { useSession } from 'next-auth/react'
+import { AlertStore } from 'stores/AlertStore'
 
 import DeleteModal from '../DeleteModal'
 import DetailsModal from '../DetailsModal'
+import { apiClient } from 'service/apiClient'
+import { authConfig } from 'service/restRequests'
+import CancelPresentationIcon from '@mui/icons-material/CancelPresentation'
+import EditIcon from '@mui/icons-material/Edit'
 
 export default function Grid() {
   const { t } = useTranslation('recurring-donation')
-  const { data }: UseQueryResult<RecurringDonationResponse[]> = useRecurringDonationList()
+  const { data }: UseQueryResult<RecurringDonationResponse[]> = useAllRecurringDonations()
   const [pageSize, setPageSize] = useState(5)
+  const { data: session } = useSession()
+  const router = useRouter()
 
   const commonProps: Partial<GridColDef> = {
     align: 'left',
     width: 150,
     headerAlign: 'left',
   }
 
+  const cancelMutation = useMutation({
+    mutationFn: async (id: string) => {
+      apiClient.get(
+        endpoints.recurringDonation.cancelRecurringDonation(id).url,
+        authConfig(session?.accessToken),
+      )
+    },
+    onError: (err) => AlertStore.show(t('common:alerts.error') + err, 'error'),
+    onSuccess: () => {
+      AlertStore.show(t('recurring-donation:alerts.cancel'), 'success')
+    },
+  })
+
+  const cancelRecurringDonation = (id: string) => {
+    if (confirm(t('recurring-donation:alerts.cancel-confirm'))) {
+      cancelMutation.mutate(id)
+    }
+  }
+
+  const editRecurringDonation = (id: string) => {
+    //open the edit page
+    router.push('/admin/recurring-donation/' + id)
```

Comment:
use the `routes` object here instead of a hardcoded string. Add it if it doesn't exist